01_Data/src/layers/sequelize/repository/Authorization.ts

```diff
@@ -26,10 +47,12 @@ export class AuthorizationRepository extends SequelizeRepository<Authorization>
         ...value.idTokenInfo,
       });
       if (authorizationModel.idTokenInfoId) {
-        let savedIdTokenInfo = (await this.s.models[IdTokenInfo.MODEL_NAME].findOne({
-          where: { id: authorizationModel.idTokenInfoId },
-          include: [{ model: IdToken, include: [AdditionalInfo] }],
-        })) as IdTokenInfo;
+        let savedIdTokenInfo = await this.idTokenInfo
+          .readAllByQuery({
```

Comment:
since we are doing:
```
.then((rows) => rows[0]);
```
should we have an interface for `this.idTokenInfo.readfirstByQuery`?

Reply:
`this.idTokenInfo.readfirstByQuery` could be a very dangerous function to exist, as it might promote usage that should break (i.e. if there should only be one variable attribute that matches and two are found, an error should be thrown).

I put a comment on line 55 explicitly to make it clear why it was safe, in this scenario, to assume one array element. It seems it wasn't clear enough; I will update this code so it breaks if there's more than one element returned.

`readOnlyByQuery`, a version of `readfirstByQuery` that throws an error if more than one entry matches the query, could be a useful function to exist. Maybe a future refactor could implement it if it seems it would simplify the code.